src/overlays/actors/ovl_En_Go2/z_en_go2.c

```diff
@@ -30,199 +216,2028 @@ const ActorInit En_Go2_InitVars = {
+                                 { 20.0f, 40.0f }, { 20.0f, 20.0f },   { 20.0f, 20.0f },   { 20.0f, 20.0f },
+                                 { 20.0f, 20.0f }, { 20.0f, 20.0f },   { 20.0f, 20.0f },   { 20.0f, 20.0f },
+                                 { 20.0f, 20.0f }, { 20.0f, 20.0f } };
+
+static struct_80034EC0_Entry sAnimations[] = {
+    { &D_06004930, 0.0f, 0.0f, -1.0f, 0x00, 0.0f },  { &D_06004930, 0.0f, 0.0f, -1.0f, 0x00, -8.0f },
+    { &D_060029A8, 1.0f, 0.0f, -1.0f, 0x00, -8.0f }, { &D_06010590, 1.0f, 0.0f, -1.0f, 0x00, -8.0f },
+    { &D_06003768, 1.0f, 0.0f, -1.0f, 0x00, -8.0f }, { &D_060038E4, 1.0f, 0.0f, -1.0f, 0x02, -8.0f },
+    { &D_06002D80, 1.0f, 0.0f, -1.0f, 0x02, -8.0f }, { &D_0600161C, 1.0f, 0.0f, -1.0f, 0x00, -8.0f },
+    { &D_06001A00, 1.0f, 0.0f, -1.0f, 0x00, -8.0f }, { &D_060021D0, 1.0f, 0.0f, -1.0f, 0x00, -8.0f },
+    { &D_06004930, 0.0f, 0.0f, -1.0f, 0x01, -8.0f }, { &D_06000750, 1.0f, 0.0f, -1.0f, 0x00, -8.0f },
+    { &D_06000D5C, 1.0f, 0.0f, -1.0f, 0x00, -8.0f }
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Go2/EnGo2_Draw.s")
+static EnGo2DustEffectData sDustEffectData[2][4] = { { { 12, 0.2f, 0.2f, 1, 18.0f, 0.0f },
+                                                       { 12, 0.1f, 0.2f, 12, 26.0f, 0.0f },
+                                                       { 12, 0.1f, 0.3f, 4, 10.0f, 0.0f },
+                                                       { 12, 0.2f, 0.2f, 1, 18.0f, 0.0f } },
+                                                     { { 12, 0.5f, 0.4f, 3, 42.0f, 0.0f },
+                                                       { 12, 0.5f, 0.4f, 3, 42.0f, 0.0f },
+                                                       { 12, 0.5f, 0.4f, 3, 42.0f, 0.0f },
+                                                       { 12, 0.5f, 0.4f, 3, 42.0f, 0.0f } } };
+
+static s32 unusedPadding[] = { 0, 0, 0 };
+
+void EnGo2_AddDust(EnGo2* this, Vec3f* pos, Vec3f* velocity, Vec3f* accel, u8 initialTimer, f32 scale, f32 scaleStep) {
+    EnGoEffect* dustEffect = this->dustEffects;
+    s16 i;
+    s16 timer;
+
+    for (i = 0; i < ARRAY_COUNT(this->dustEffects); i++, dustEffect++) {
+        if (dustEffect->type != 1) {
+            dustEffect->scale = scale;
+            dustEffect->scaleStep = scaleStep;
+            if (1) {}
+            timer = initialTimer;
+            dustEffect->timer = timer;
+            dustEffect->type = 1;
+            dustEffect->initialTimer = initialTimer;
+            dustEffect->pos = *pos;
+            dustEffect->accel = *accel;
+            dustEffect->velocity = *velocity;
+            return;
+        }
+    }
+}
+
+void EnGo2_UpdateDust(EnGo2* this) {
+    EnGoEffect* dustEffect = this->dustEffects;
+    f32 randomNumber;
+    s16 i;
+
+    for (i = 0; i < ARRAY_COUNT(this->dustEffects); i++, dustEffect++) {
+        if (dustEffect->type) {
+            dustEffect->timer--;
+
+            if (dustEffect->timer == 0) {
```

Comment:
Does this work?
```suggestion
            if (--dustEffect->timer == 0) {
```

Reply:
idk if OoTR has a preference on this, but personally I prefer increment/decrement operations to be on their own line whenever possible.

Reply:
Okay yeah I just double checked the repo and decrementing on the previous line is indeed the preferred standard!